doc/dev_guide/coding_style.rst

```diff
@@ -35,3 +35,31 @@ adding a ``post-commit`` file to ``.git/hook/`` with the following content:
     GIT_COMMITTER_NAME="black" GIT_COMMITTER_EMAIL="black@email.com" git
     commit --author="black <black@email.com>" -m "Automatic style
     corrections courtesy of black"
+
+Deprecations
+============
+Hyperspy follows `semetic versioning <https://semver.org>`_ where changes follow such that:
+
+1. MAJOR version when you make incompatible API changes
+2. MINOR version when you add functionality in a backward compatible manner
+3. PATCH version when you make backward compatible bug fixes
+
+This means that as little, ideally no, functionality should break between minor releases.
+Deprecation warnings are raised whenever possible and feasible for functions/methods/properties/arguments,
+so that users get a heads-up one (minor) release before something is removed or changes, with a possible
+alternative to be used.
+
+The decorator should be placed right above the object signature to be deprecated:
+
+@deprecated(since=1.7.0, removal=2.0.0, alternative="bar")
+def foo(self, n):
+    return n + 1
+
+@deprecated_argument(since=1.7.0, removal=2.0.0,name="x", alternative="y")
+def this_property(y):
+    return y
+
+This will update the docstring, and print a visible deprecation warning telling the user to user the
+alternative function or argument.
+
+These deprecation wrappers are copied from work by Håkon Wiik Ånes and kikchipy.
```

Comment:
Sounds good :) I'll just leave that they came from kikuchipy.

I do think they are quite a good way to handle deprecations and it might be good to promote them for wider usage.  If someone takes them from hyperspy it would be good to have the credit make it's way back to you!

Reply:
Thanks, that is more than enough (: